paddle/fluid/operators/distributed/proto_encoder_helper.h

```diff
@@ -83,7 +83,9 @@ class ProtoEncodeHelper {
 
   ~ProtoEncodeHelper() {
     // Make sure callers didn't do operations that went over max_size promised
-    PADDLE_ENFORCE_LE(p_, limit_);
+#define REPLACE_ENFORCE_GLOG 1
```

Comment:
`#ifdef`?

Reply:
here is #define, because we should force to not throw exception here

Reply:
How about put `throw_on_error` directly.

Reply:
```throw_on_error``` will detect the ```REPLACE_ENFORCE_GLOG``` flag and still throw a exception if we use it directly